File: Leetcode/C/1-bit and 2-bit Characters.c
bool isOneBitCharacter(int* bits, int bitsSize) {
    if(bitsSize == 1)
        return 1;
    
    int i = bitsSize - 2;
    int sum = 0;
    
    while(bits[i] == 1 && i>=0){
        sum += 1;
        i--;
    }
    
    return (sum-1)% 2;
}

File: Leetcode/python/1-bit and 2-bit Characters.py
class Solution:
    def isOneBitCharacter(self, bits):
        """
        :type bits: List[int]
        :rtype: bool
        """
        i=0
        while i< len(bits) - 2:
            if(bits[i] == 1):
                i += 2
            else:
                i += 1
        
        if i == len(bits) - 1:
            return True
        #else i == len(bits) - 2
        elif bits[i] == 0:
            return True 
        else: 
            return False
            